kexec-tools: Fix building on x86_64 with binutils 2.41

namiltd opened the pull request:
Newer versions of the GNU assembler (observed with binutils 2.41) will complain about the ".arch i386" in files assembled with "as --64", with the message "Error: 64bit mode not supported on 'i386'".

Fix by moving ".arch i386" below the relevant ".code32" directive, so that the assembler is no longer expecting 64-bit instructions to be used by the time that the ".arch i386" directive is encountered.

Based on similar iPXE fix:
ipxe/ipxe@6ca597eee

Cherry picked from commit https://github.com/horms/kexec-tools/commit/328de8e
Fixes current build issues on Github checks, e.g.: https://github.com/openwrt/openwrt/actions/runs/9416868485/job/25940931754

neheb commented on package/boot/kexec-tools/patches/001-fix_x86_64_building.patch:
```diff
@@ -0,0 +1,89 @@
+ purgatory/arch/i386/setup-x86.S        | 2 +-
+ 4 files changed, 4 insertions(+), 4 deletions(-)
+
+diff --git a/purgatory/arch/i386/entry32-16-debug.S b/purgatory/arch/i386/entry32-16-debug.S
```
refresh patches please

@Ansuel why isn't CI working?

guidosarducci replied:
CI is broken for lots of people's PR's I believe. See here: https://github.com/openwrt/openwrt/pull/15639#issuecomment-2154307314.

#15659 should address both problems with `kexec-tools`.

namiltd replied:
@naheb Done, patches refreshed.

namiltd committed:
kexec-tools: Fix building on x86_64 with binutils 2.41

Newer versions of the GNU assembler (observed with binutils 2.41) will
complain about the ".arch i386" in files assembled with "as --64",
with the message "Error: 64bit mode not supported on 'i386'".

Fix by moving ".arch i386" below the relevant ".code32" directive, so
that the assembler is no longer expecting 64-bit instructions to be used
by the time that the ".arch i386" directive is encountered.

Based on similar iPXE fix:
ipxe/ipxe@6ca597eee

Signed-off-by: Michel Lind <michel@michel-slm.name>
Signed-off-by: Simon Horman <horms@kernel.org>
Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
(cherry picked from commit horms/kexec-tools@328de8e)

hauke commented:
Please do not use cherry pick when it is coming from a different repository. 
Please do not copy signed-off-by lines from the patch you are adding, only if these people actually approved your OpenWrt change.

hauke commented:
I merged the similar PR #15659 instead.
